configserver/src/main/java/com/yahoo/vespa/config/server/application/ConfigConvergenceChecker.java

```diff
@@ -108,46 +117,67 @@ public JSONResponse getServiceConfigGenerationResponse(Application application,
 
     @Override
     public void deconstruct() {
-        clientBuilderFactory.close();
-    }
-
-    @Path(statePath)
-    public interface StateApi {
-        @Path(configSubPath)
-        @GET
-        JsonNode config();
-    }
-
-    public interface StateApiFactory {
-        StateApi createStateApi(Client client, URI serviceUri);
+        try {
+            client.close();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
     }
 
     /** Gets service generation for a list of services (in parallel). */
     private Map<ServiceInfo, Long> getServiceGenerations(List<ServiceInfo> services, Duration timeout) {
-        return services.parallelStream()
-                       .collect(Collectors.toMap(service -> service,
-                                                 service -> {
-                                                     try {
-                                                         return getServiceGeneration(URI.create("http://" + service.getHostName()
-                                                                                                + ":" + getStatePort(service).get()), timeout);
-                                                     }
-                                                     catch (ProcessingException e) { // Cannot connect to service to determine service generation
-                                                         return -1L;
-                                                     }
-                                                 },
-                                                 (v1, v2) -> { throw new IllegalStateException("Duplicate keys for values '" + v1 + "' and '" + v2 + "'."); },
-                                                 LinkedHashMap::new
-                                                ));
+        List<CompletableFuture<Void>> inprogressRequests = new ArrayList<>();
+        ConcurrentMap<ServiceInfo, Long> temporaryResult = new ConcurrentHashMap<>();
+        for (ServiceInfo service : services) {
+            int statePort = getStatePort(service).orElse(0);
+            if (statePort <= 0) continue;
+
+            URI uri = URI.create("http://" + service.getHostName() + ":" + statePort);
+            CompletableFuture<Void> inprogressRequest = getServiceGeneration(uri, timeout)
+                    .handle((result, error) -> {
+                        if (result != null) {
+                            temporaryResult.put(service, result);
+                        } else {
+                            log.log(
+                                    LogLevel.DEBUG,
```

Comment:
`Level.FINE`

Reply:
I think they are equivalent (`LogLevel` being the Vespa wrapper for `Level`). Any benefits using `Level` instead?

Reply:
I know @gjoranv went to great lengths to replace with `Level` values some time ago. I don't know why, though.

Reply:
Ok, I removed use of `LogLevel` in new PR.

Reply:
I had to remove it from config in order to use config inside `vespalog`, and then it was proposed to remove `LogLevel` usage from all modules. I think the motivation was that custom log levels are unnecessary and a bit odd.